src/components/GroupCallButton.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import { useUserInfoStore } from "@/hooks/useUserInfoStore";
+import SendBirdCall from "sendbird-calls";
+import { useMutation } from "~/liveblocks.config";
+import { useSendBirdInit } from "@/hooks/useSendBirdCall";
+import { memo, useRef } from "react";
+
+interface TypeGroupCallId {
+  roomId: string;
+}
+
+const GroupCallButton = memo((groupCallId: TypeGroupCallId) => {
+  const userInfo = useUserInfoStore();
+  const audioRef = useRef<HTMLAudioElement>(null);
+
+  const roomParams = {
+    roomType: SendBirdCall.RoomType.LARGE_ROOM_FOR_AUDIO_ONLY,
+  };
+
+  const enterParams = {
+    videoEnabled: false, // 비디오 미사용
+    audioEnabled: true, // 오디오만 출력
+  };
+
+  // 프로젝트 내 그룹 콜 ID 설정
+  const updateRoomId = useMutation(({ storage }, roomId: string) => {
+    const groupCallId = storage.get("groupCallId");
+    groupCallId.set("roomId", roomId);
+  }, []);
+
+  const authOption = { userId: userInfo._id, accessToken: userInfo.token }; // Authentication options using user information
+  useSendBirdInit(authOption);
+
+  const joinGroupCall = async () => {
+    // 아직 프로젝트 내에 그룹 콜이 진행되지 않은 경우, 새로운 방 생성
+    if (!groupCallId.roomId) {
+      SendBirdCall.createRoom(roomParams).then((room) => {
+        updateRoomId(room.roomId); // 프로젝트 내 그룹 콜 ID 설정
+        console.log("방 생성 성공, roomId: ", room.roomId);
+      });
+    }
+
+    SendBirdCall.fetchRoomById(groupCallId.roomId).then((room) => {
+      console.log("방 불러오기 성공: ", room);
+
+      if (
+        room.participants.filter(
+          (participant) => participant.user.userId === userInfo._id
+        ).length > 0
```

Comment:
```suggestion
      room.participants.some(participant => participant.user.userId === userInfo._id))

```
자주 쓰이지 않아서 깜빡하셨겠지만 자바스크립트 메서드중 some을 활용하면 이런 경우의 로직을 작성하실 때 훨씬 예쁘게 짤 수 있습니다!

Reply:
오.. 하나라도 조건을 만족하면 참을 반환하는 메소드네요. 몰랐습니다ㅜㅠ 이렇게 하면 전체를 다 필터링하고, 길이를 비교하는 과정을 생략하고 하나라도 조건을 만족하면 연산이 멈추니 성능적으로도 미세하게 도움이 되겠네요